packages/api/src/external/carequality/document/document-query-retrieval.ts

```diff
@@ -0,0 +1,65 @@
+import { PurposeOfUse } from "@metriport/shared";
+import { DocumentRetrievalReqToExternalGW, DocumentReference } from "@metriport/ihe-gateway-sdk";
+import dayjs from "dayjs";
+import { DocumentWithMetriportId } from "./shared";
+import { DocumentQueryResult } from "../document-query-result";
+import { Organization } from "@metriport/core/domain/organization";
+
+const SUBJECT_ROLE_CODE = "106331006";
+const SUBJECT_ROLE_DISPLAY = "Administrative AND/OR managerial worker";
+
+export function createCQDocumentRetrievalRequests({
+  requestId,
+  cxId,
+  organization,
+  documentReferences,
+  documentQueryResults,
+}: {
+  requestId: string;
+  cxId: string;
+  organization: Organization;
+  documentReferences: DocumentWithMetriportId[];
+  documentQueryResults: DocumentQueryResult[];
+}): DocumentRetrievalReqToExternalGW[] {
+  const orgOid = organization.oid;
+  const orgName = organization.data.name;
+  const user = `${orgName} System User`;
+  const now = dayjs().toISOString();
+
+  const requests: DocumentRetrievalReqToExternalGW[] = documentQueryResults.map(
+    documentQueryResult => {
+      const { patientId, gateway } = documentQueryResult.data;
+
+      const requestDocReferences: DocumentReference[] = documentReferences.filter(
+        docRef => docRef.homeCommunityId === gateway?.homeCommunityId
+      );
+
+      return {
+        id: requestId,
+        cxId: cxId,
+        patientId: patientId,
+        timestamp: now,
+        samlAttributes: {
+          subjectId: user,
+          subjectRole: {
+            code: SUBJECT_ROLE_CODE,
+            display: SUBJECT_ROLE_DISPLAY,
+          },
+          organization: orgName,
+          organizationId: orgOid,
+          homeCommunityId: orgOid,
+          purposeOfUse: PurposeOfUse.TREATMENT,
+        },
+        gateway: {
+          homeCommunityId: gateway?.homeCommunityId || "",
+          url: gateway?.url || "",
```

Comment:
when would we ever not have a gateway here?

should we throw an error if `gateway` is falsey? or skip over the item in the mapping?

I don't see a situation where we'd ever be sending a request without the `homeCommunityId` or `url`, since the IHE GW can't do anything with it

Reply:
Good point ill filter it out if it does appear empty